File: 251126/계속 중첩되는 사각형/continuously-overlapping-squares.py
n = int(input())
x1, y1, x2, y2 = [], [], [], []
for _ in range(n):
    a, b, c, d = map(int, input().split())
    x1.append(a)
    y1.append(b)
    x2.append(c)
    y2.append(d)

OFFSET = 100
arr = [[0] * (OFFSET * 2 + 1) for _ in range(OFFSET * 2 + 1)]

for i in range(x1[0], x2[0]):
    for j in range(y1[0], y2[0]):
        arr[i + OFFSET][j + OFFSET] = 1

for i in range(x1[1], x2[1]):
    for j in range(y1[1], y2[1]):
        arr[i + OFFSET][j + OFFSET] = 2

count = 0
for i in range(len(arr)):
    for j in range(len(arr)):
        if arr[i][j] == 2:
            count += 1

print(count)

File: 260319/계속 중첩되는 사각형/continuously-overlapping-squares.js
const OFFSET = 100;
const MAX_SIZE = 201

const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

let grid = Array.from({ length: MAX_SIZE }, () => Array(MAX_SIZE).fill(''));

const n = Number(input[0]);
const rectangles = [];

for (let i = 1; i <= n; i++) {
    let [x1, y1, x2, y2] = input[i].split(' ').map(Number);

    x1 += OFFSET;
    y1 += OFFSET;
    x2 += OFFSET;
    y2 += OFFSET;

    rectangles.push([x1, y1, x2, y2]);
}

for (let i = 0; i < n; i++) {
    let [x1, y1, x2, y2] = rectangles[i];

    if (i % 2 === 1) { // 첫 번째가 인덱스 0이니까 1이어야 파란색
        for (let x = x1; x < x2; x++) {
            for (let y = y1; y < y2; y++) {
                grid[x][y] = 'B'
            }
        }
    } else {
        for (let x = x1; x < x2; x++) {
            for (let y = y1; y < y2; y++) {
                grid[x][y] = 'R'
            }
        }
    }
}

let answer = 0;
for (let i = 0; i < MAX_SIZE; i++) {
    for (let j = 0; j < MAX_SIZE; j++) {
        if (grid[i][j] === 'B') {
            answer += 1;
        }
    }
}

console.log(answer);